Makefile

```diff
@@ -82,6 +82,7 @@ else
 endif
 
 LDFLAGS := $(LDFLAGS) -X "main.MakeVersion=$(MAKE_VERSION)" -X "main.Version=$(GITEA_VERSION)" -X "main.Tags=$(TAGS)"
+CGO_CFLAGS ?= $(shell $(GO) env CGO_CFLAGS) -DSQLITE_MAX_VARIABLE_NUMBER=32766
```

Comment:
```suggestion
CGO_EXTRA_CFLAGS := -DSQLITE_MAX_VARIABLE_NUMBER=32766
CGO_CFLAGS ?= $(shell $(GO) env CGO_CFLAGS) $(CGO_EXTRA_CFLAGS)
```

Reply:
Also, because this depends on the existance of `go`, we might want to move the `go env` call inside the `HAS_GO` conditional further above (probably means to move the flags initialization before that check.

This is important because otherwise the `go-check` target can not print a proper warning if `go` isn't present.